api/v1alpha1/applicationset_types.go

```diff
@@ -342,6 +346,10 @@ type SCMProviderGeneratorFilter struct {
 	LabelMatch *string `json:"labelMatch,omitempty"`
 	// A regex which must match the branch name.
 	BranchMatch *string `json:"branchMatch,omitempty"`
+	// A regex which must match the pull request tile.
+	PullRequestBranchMatch *string `json:"pullRequestTitleMatch,omitempty"`
```

Comment:
Should there be two filters, one to match source branches, and the other to match target branches?

Reply:
That's an interesting point! I can't recall ever seeing a source branch being merged into a non-default target branch (main, master etc.). In addition, usually, all source branches get merged into one single target branch (otherwise you got a very messy repo 😄 ) which makes filtering the target branch pointless. So IMHO this change would only support a corner case. What do you think?

Reply:
Consider this use case: I want to spin up test environments for each PR against `main`. Developer A opens a PR from `feature/whatever` to `main`. Developer B has suggestions but instead of writing them all out, just opens a PR from `feature/whatever-review` to `feature/whatever`. 

I think this is common enough to justify giving the AppSet the designer to avoid creating an App for that second PR.

Reply:
But if it ends up being a lot of work, we could certainly leave it for a future enhancement.